lib/functions/general/git.sh

```diff
@@ -56,9 +56,7 @@ function git_ensure_safe_directory() {
 	if [[ -n "$(command -v git)" ]]; then
 		local git_dir="$1"
 		display_alert "git: Marking all directories as safe, which should include" "$git_dir" "debug"
-		if ! grep -q "directory = \*" "${HOME}/.gitconfig" 2> /dev/null; then
-			git config --global --add safe.directory "*"
-		fi
+		regular_git -C "$1" config --local --get safe.directory "$1" > /dev/null || regular_git -C "$1" config --local --add safe.directory "$1"
```

Comment:
@igorpecovnik IMO it would be more consistent to replace `"$1"` by the local `"$git_dir"` in all places? That would leverage readability and mitigate risk of overlooking something in case of future changes.